### DISCUSSION TITLE
Read Json Object from HTTPEndpoint Incoming request from another API

### DISCUSSION THREAD

**Discussion by Rathna314** [elsa-workflows/elsa-core#4996] [Q&A]
How do I read the json sent from another API to ELsa Workflow V2 Dashboard HTTPEndPoint. Need some sample in Json.

**Comment by glime-ai** [comment]
To read JSON sent from another API to an Elsa Workflow V2 Dashboard HTTP Endpoint, you would typically use the `HttpEndpoint` activity to set up an endpoint that can receive HTTP requests. The JSON payload sent to this endpoint can then be accessed within the workflow.

Here's a basic example of how you might define a workflow that reads JSON from an incoming HTTP POST request:

1. Add an `HttpEndpoint` activity to your workflow with the following settings:
   - Method: POST
   - Path: /your-endpoint-path (e.g., /api/data)

2. Add a `RunJavaScript` or similar activity to process the incoming JSON. You can access the JSON payload using the `input` variable, which represents the incoming request content.

3. Use expressions to access the JSON properties. For example, if the JSON payload is `{"name": "Elsa", "age": 3}`, you can access the `name` property with an expression like `input.name`.

Here's a sample JSON representation of such a workflow:

```json
{
  "name": "ReceiveJsonWorkflow",
  "activities": [
    {
      "id": "http-endpoint",
      "type": "HttpEndpoint",
      "properties": {
        "Method": "POST",
        "Path": "/api/data"
      }
    },
    {
      "id": "process-json",
      "type": "RunJavaScript",
      "properties": {
        "Script": "return { name: input.name, age: input.age };"
      }
    }
  ],
  "connections": [
    {
      "sourceId": "http-endpoint",
      "targetId": "process-json"
    }
  ]
}
```

To execute this workflow, you would send a POST request to `https://localhost:5001/workflows/api/data` with a JSON payload.

Here's a sample `curl` command to send a JSON payload to the workflow endpoint:

```bash
curl --location --request POST 'https://localhost:5001/workflows/api/data' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "Elsa",
    "age": 3
}'
```

Please note that the actual endpoint URL and port may vary based on your Elsa Dashboard setup.

For more detailed examples and documentation, you can refer to the Elsa Workflow documentation and samples provided in the context section, specifically the HTTP Backend API and HTTP Post Users API samples.

**Comment by Rathna314** [comment]
Is "type": "RunJavaScript", available in Version 2.13 Dashboard. If not which activity should I use.

**Comment by Rathna314** [comment]
Can someone respond please. Is "type": "RunJavaScript", available in Version 2.13 Dashboard. If not which activity should I use. I am unable to see RunJavaScript in 2.13 version.

**Comment by sfmskywalker** [comment]
Yes, but it's not added to the system by default. To make it available, you need to call `AddJavaScriptActivities` on the elsa options builder.

**Comment by sfmskywalker** [comment]
[This guide](https://v2.elsaworkflows.io/docs/guides/document-approval#http-endpoint) covers a similar scenario where it demonstrates reading a payload sent to an HTPT Endpoint activity.

**Comment by glime-ai** [comment]
To read JSON sent from another API to an Elsa Workflow V2 Dashboard HTTP Endpoint, you would typically use the `HttpEndpoint` activity to create an endpoint that can receive HTTP requests. When a request is received, the JSON payload can be accessed and processed within the workflow.

Here's a basic example of how you might define a workflow in JSON format that uses the `HttpEndpoint` activity to receive and read JSON data:

```json
{
  "name": "ReceiveJsonWorkflow",
  "activities": [
    {
      "name": "HttpEndpoint",
      "type": "HttpEndpoint",
      "properties": {
        "Path": "/api/data",
        "Method": "POST",
        "ReadContent": true
      },
      "outputs": {
        "ReceivedData": {
          "type": "Json",
          "expression": "activity.Output"
        }
      }
    },
    {
      "name": "HandleReceivedData",
      "type": "RunJavaScript",
      "properties": {
        "Script": "return JSON.parse(receivedData);",
        "Variables": {
          "receivedData": "{{ Activities.HttpEndpoint.ReceivedData }}"
        }
      }
    }
  ],
  "connections": [
    {
      "sourceActivityId": "HttpEndpoint",
      "targetActivityId": "HandleReceivedData",
      "outcome": "Done"
    }
  ]
}
```

In this example, the `HttpEndpoint` activity is configured to listen for POST requests at the path `/api/data`. It's set to read the content of the request, which should be in JSON format. The received JSON data is then passed to a `RunJavaScript` activity, which can process the JSON data as needed.

Please note that the `RunJavaScript` activity is used here as an example of how you might process the JSON data within the workflow. Depending on your version of Elsa and the activities available, you might need to use a different activity or approach to process the JSON data.

If you're using Elsa Dashboard and want to create this workflow programmatically, you would import this JSON definition into the Elsa Dashboard or use the Elsa API to create the workflow definition.

Remember to replace the placeholder values with actual values that match your use case, and ensure that the `HttpEndpoint` activity is properly secured if it's exposed to the public internet.
